benches/benches/block_target_gas.rs

```diff
@@ -138,6 +154,209 @@ fn run(
     });
 }
 
+/// Sets up a service with a full database. Returns the service with the associated Runtime.
+/// The size of the database can be overridden with the `STATE_SIZE` environment variable.
+fn service_with_contract_id(
+    contract_id: ContractId,
+) -> (fuel_core::service::FuelService, tokio::runtime::Runtime) {
+    let rt = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap();
+    let _drop = rt.enter();
+    let mut database = Database::rocksdb();
+    let mut config = Config::local_node();
+    config
+        .chain_conf
+        .consensus_parameters
+        .tx_params
+        .max_gas_per_tx = TARGET_BLOCK_GAS_LIMIT;
+    config.chain_conf.initial_state.as_mut().unwrap().contracts =
+        Some(vec![ContractConfig {
+            contract_id,
+            code: vec![],
+            salt: Default::default(),
+            state: None,
+            balances: None,
+            tx_id: None,
+            output_index: None,
+            tx_pointer_block_height: None,
+            tx_pointer_tx_idx: None,
+        }]);
+
+    config
+        .chain_conf
+        .consensus_parameters
+        .predicate_params
+        .max_gas_per_predicate = TARGET_BLOCK_GAS_LIMIT;
+    config.chain_conf.block_gas_limit = TARGET_BLOCK_GAS_LIMIT;
+    config.utxo_validation = false;
+    config.block_production = Trigger::Instant;
+
+    // Override state size if the env var is set
+    let state_size_env_var = "STATE_SIZE";
+    let state_size = if let Some(value) = std::env::var_os(state_size_env_var) {
+        let value = value.to_str().unwrap();
+        let value = value.parse::<u64>().unwrap();
+        println!("Overriding state size with {}", value);
+        value
+    } else {
+        STATE_SIZE
+    };
```

Comment:
NIt: I'd like a more compact approach here, maybe something like

```suggestion
    let state_size = std::env::var_os("STATE_SIZE").map(|value| {
        let value = value.to_str().unwrap();
        let value = value.parse::<u64>().unwrap();
        println!("Overriding state size with {}", value);
        value
    }).unwrap_or(STATE_SIZE);
```